src/nos/core/orchestrator.py

```diff
@@ -184,3 +188,134 @@ async def process_file_registered_notification(self, *, file_id: str):
             notification=notifications.FILE_REGISTERED_TO_DS.formatted(file_id=file_id),
         )
         log.info("Sent File Upload Completed notification to data steward")
+
+    async def _iva_code_requested(self, *, user: User):
+        """Send notifications relaying that an IVA code has been requested.
+
+        One notification is sent to the user to confirm that their request was received.
+
+        Another notification is sent to the data steward to inform them of the request.
+        """
+        # Send a notification to the user
+        await self._notification_emitter.notify(
+            email=user.email,
+            full_name=user.name,
+            notification=notifications.IVA_CODE_REQUESTED_TO_USER,
+        )
+
+        # Send a notification to the data steward
+        await self._notification_emitter.notify(
+            email=self._config.central_data_stewardship_email,
+            full_name=DATA_STEWARD_NAME,
+            notification=notifications.IVA_CODE_REQUESTED_TO_DS.formatted(
+                full_user_name=user.name, email=user.email
+            ),
+        )
+
+    async def _iva_code_transmitted(self, *, user: User):
+        """Send a notification that an IVA code has been transmitted to the user."""
+        await self._notification_emitter.notify(
+            email=user.email,
+            full_name=user.name,
+            notification=notifications.IVA_CODE_TRANSMITTED_TO_USER,
+        )
+
+    async def _iva_code_submitted(self, *, user: User):
+        """Send a notification to the data steward that an IVA code has been submitted."""
```

Comment:
```suggestion
    async def _iva_code_validated(self, *, user: User):
        """Send a notification to the data steward that an IVA code has been validated."""
```

When a wrong validation code is submitted, no notification is sent (unless it happens too many times).

Reply:
Changed